core/server/api_container/server/startosis_engine/kurtosis_types/service_config/ready_condition.go

```diff
@@ -109,17 +109,16 @@ func (readyCondition *ReadyCondition) GetRecipe() (recipe.Recipe, *startosis_err
 
 	httpRecipe, found, interpretationErr := kurtosis_type_constructor.ExtractAttrValue[recipe.HttpRequestRecipe](readyCondition.KurtosisValueTypeDefault, RecipeAttr)
 	genericRecipe = httpRecipe
-	if !found {
-		return nil, startosis_errors.NewInterpretationError("Required attribute '%s' could not be found on type '%s'",
-			RecipeAttr, ReadyConditionTypeName)
-	}
 	//TODO we should rework the recipe types to inherit a single common type, this will avoid the double parsing here.
 	if interpretationErr != nil {
 		execRecipe, _, interpretationErr := kurtosis_type_constructor.ExtractAttrValue[*recipe.ExecRecipe](readyCondition.KurtosisValueTypeDefault, RecipeAttr)
 		if interpretationErr != nil {
 			return nil, interpretationErr
 		}
 		genericRecipe = execRecipe
+	} else if !found {
```

Comment:
I wonder if we could refactor this code a bit so we reduce how nested this is

There are three cases:
1. Parse a HTTP recipe, if `httpInterpretationErr == nil`, we can safely return. (eject early)
2. Parse an Exec recipe, if `execInterpretationErr == nil`, we can safely return. (eject early)
3. If the code wasn't ejected, return an error at the end of the func

Reply:
@victorcolombo I updated `ReadyCondition.GetRecipe()` to use the following logic to reduce nesting.  Let me know if it looks good.

1. Parse a HTTPRecipe, if `found == false` and `err == nil`: return an error saying that the required attribute does not exist. If `err == nil`: return the HTTPRecipe.
2. Parse an ExecRecipe, if `err == nil`: return the ExecRecipe.
3. Return `err`

Reply:
Chiming in here as I was debugging something similar for Derek today. I think the issue was actually in the `ExtractAttrValue` function, it was returning `found = false` even if the attribute was found, but the type was incorrect (https://github.com/kurtosis-tech/kurtosis/pull/514/files). Which I guess could be discussed philosophically, but overall I believe with my quick fix you shouldn't need this extra logic here.

Reply:
Thanks @gbouv!  I simplified the first condition based on this change.